lib/webdriverCommands.ts

```diff
@@ -0,0 +1,91 @@
+class Endpoint {
+  constructor(private pattern: string, private method: HttpMethod, public name: CommandName) {}
+
+  matches(url, method) {
+    let urlParts = url.split('/');
+    let patternParts = this.pattern.split('/');
+
+    if (method != this.method || urlParts.length != patternParts.length) {
+      return false;
+    }
+    for (let idx in patternParts) {
+      if (!patternParts[idx].startsWith(':') && patternParts[idx] != urlParts[idx]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  getParams(url) {
+    let urlParts = url.split('/');
+    let patternParts = this.pattern.split('/');
+
+    let params = {};
+    for (let idx in patternParts) {
+      if (patternParts[idx].startsWith(':')) {
+        let paramName = patternParts[idx].slice(1);
+        params[paramName] = urlParts[idx];
+      }
+    }
+    return params;
+  }
+}
+
+/**
+ * Returns a new WebdriverCommand object for the resource at the given URL.
+ */
+export function parseWebDriverCommand(url, method, data: string) {
+  let parsedData = {};
+  if (data) {
+    parsedData = JSON.parse(data);
+  }
+
+  for (let endpoint of endpoints) {
+    if (endpoint.matches(url, method)) {
+      let params = endpoint.getParams(url);
+      return new WebDriverCommand(endpoint.name, params, parsedData);
+    }
+  }
+
+  return new WebDriverCommand(CommandName.UNKNOWN, {}, {'url': url});
+}
+
+let endpoints: Endpoint[] = [];
```

Comment:
Done. I'm not super happy with how this WebDriver parsing stuff is done, I'll probably change things around when I add a few more commands.